derive/src/impls/columns.rs

```diff
@@ -0,0 +1,135 @@
+use quote::quote;
+use syn::{Data, DeriveInput, Result};
+
+use crate::common::{ensure, is_repr_c};
+
+/// Implements `Borrow`, `BorrowMut`, `From`, `Index`, `IndexMut`, and
+/// `Default`.
+pub(crate) fn try_derive(ast: DeriveInput) -> Result<proc_macro2::TokenStream> {
+    let is_struct = matches!(ast.data, Data::Struct(_));
+    ensure!(is_struct, &ast, "expected `struct`");
+
+    // Check that the struct is `#[repr(C)]`
+    let repr_c = is_repr_c(&ast.attrs);
+    ensure!(repr_c, &ast, "column struct must be `#[repr(C)]`");
+
+    // The name of the struct.
+    let name = &ast.ident;
+
+    // Safety: `u8` is guaranteed to have a `size_of` of 1.
+    // https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout
+    let num_columns = quote!(::core::mem::size_of::<#name<u8>>());
```

Comment:
@Nashtare I am not sure to understand this. If you have `OpsColumnsView<u32>`, and you cast it to array of its elements, number of columns (elements in the array) should be size_of::<u8>/size_of<u32>?